packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -433,9 +503,10 @@ export abstract class SharedObject<TEvent extends ISharedObjectEvents = ISharedO
      * For messages from a remote client, this will be undefined.
      */
     private process(message: ISequencedDocumentMessage, local: boolean, localOpMetadata: unknown) {
-        this.emit("pre-op", message, local, this);
+        this.verifyNotClosed(); // This will result in container closure.
+        this.safeRaiseEvent("pre-op", message, local, this);
```

Comment:
Do we have a work item to get rid of pre-op entirely?  I think prosemirror is the only thing still using it?

Reply:
See my comment below, that's not the issue I'm after (even though I agree with you fully here)